### DISCUSSION TITLE
How to make the compiler for TS not emit everything to be undefined

### DISCUSSION THREAD

**Discussion by OmGuptaIND** [bufbuild/buf#2701] [Q&A]
From proto v3, everything can be undefined, which is great for general use cases. Still, for our use cases, proto2 and the definition of required and optional makes a ton of sense, which decreases the heavy load on us to parse and make sure everything is all right; I know this is what protobuf was tried to navigate from or make it agnostic.

But I am confident for our use case, proto 2 makes the most sense, soo here is my question: How do I compile the proto file, which all have required and optional written in them, to a TS file, which Buf does beautifully but also make sure it doesn't end up making everything undefined. 

Thanks a ton, guys, for reading. Really appreciate it.

**Comment by timostamm** [comment]
@OmGuptaIND, do you use [protobuf-es](https://github.com/bufbuild/protobuf-es)? We have an issue to track support for proto2 required here: https://github.com/bufbuild/protobuf-es/issues/414
